### ISSUE TITLE
Unable to upgrade to 0.56.0 from 0.55.4

### ISSUE THREAD

**Issue by yeomann** [facebook/react-native#20273]
- [x] Review the documentation: https://facebook.github.io/react-native
  - [x] Search for existing issues: https://github.com/facebook/react-native/issues
  - [x] Use the latest React Native release: https://github.com/facebook/react-native/releases

## Environment
  OS: macOS High Sierra 10.13.6
  Node: 9.11.1
  Yarn: 1.7.0
  npm: 6.1.0
  Watchman: 4.9.0
  Xcode: Xcode 9.4.1 Build version 9F2000
  Android Studio: 3.1 AI-173.4720617

Packages: (wanted => installed)
  react: 16.3.1 => 16.3.1
  react-native: 0.55.4 => 0.55.4

## Description

*Try 1:*
I have created new folder and installed with new version 0.56.0 with success. but when I tried to do same on my production ready code, which is coded with 0.55.4 I am having hard time. I have tried manual way of upgrading it messes up everything. it renamed whole project and created a huge mess up bcz of directories renaming. (I think this because I have in something like this "name-string-something" in `package.json` for "name" ) I tried to clear all mess up & tried to run which resulted in failure. 
*Try2:*
I have created a new branch of my code, I tried first to just upgrade Babel to 7 using `yarn upgrade babel-preset-react-native@next` and tried to run I see this error below in picture.
<img width="305" alt="screen shot 2018-07-18 at 12 43 06" src="https://user-images.githubusercontent.com/5970751/42874297-6cef5df2-8a89-11e8-9f53-0129d45924c2.png">

then I thought maybe I should do this in `index.js`
```
AppRegistry.registerComponent("ispApp", () => require("./src/App").default);
```

which didn't help either. I thought maybe I need to upgrade react native so I tried `react-native-git-upgrade` which resulted pretty much same error

```
git-upgrade info Configure Git environment
git-upgrade info Init Git repository
git-upgrade info Add all files to commit
git-upgrade info Commit current project sources
git-upgrade info Create a tag before updating sources
git-upgrade info Generate old version template
git-upgrade ERR! An error occurred during upgrade:
git-upgrade ERR! Error: Plugin 0 specified in "/Users/yeoman/Development/Xeront/ReactNative/ispApp/node_modules/babel-preset-react-native/index.js" provided an invalid property of "default" (While processing preset: "/Users/yeoman/Development/Xeront/ReactNative/ispApp/node_modules/babel-preset-react-native/index.js")
    at Plugin.init (/usr/local/lib/node_modules/react-native-git-upgrade/node_modules/babel-core/lib/transformation/plugin.js:131:13)
    at Function.normalisePlugin (/usr/local/lib/node_modules/react-native-git-upgrade/node_modules/babel-core/lib/transformation/file/options/option-manager.js:152:12)
    at /usr/local/lib/node_modules/react-native-git-upgrade/node_modules/babel-core/lib/transformation/file/options/option-manager.js:184:30
    at Array.map (<anonymous>)
    at Function.normalisePlugins (/usr/local/lib/node_modules/react-native-git-upgrade/node_modules/babel-core/lib/transformation/file/options/option-manager.js:158:20)
    at OptionManager.mergeOptions (/usr/local/lib/node_modules/react-native-git-upgrade/node_modules/babel-core/lib/transformation/file/options/option-manager.js:234:36)
    at /usr/local/lib/node_modules/react-native-git-upgrade/node_modules/babel-core/lib/transformation/file/options/option-manager.js:265:14
    at /usr/local/lib/node_modules/react-native-git-upgrade/node_modules/babel-core/lib/transformation/file/options/option-manager.js:323:22
    at Array.map (<anonymous>)
    at OptionManager.resolvePresets (/usr/local/lib/node_modules/react-native-git-upgrade/node_modules/babel-core/lib/transformation/file/options/option-manager.js:275:20)
git-upgrade ERR! Restore initial sources
Note: checking out 'project-snapshot'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 44b670a... Project snapshot
```

I have been trying from more than a day now. my upgrade seems no where close. need help please. 
Here is my package.json
```
{
    "name": "ispApp",
    "version": "0.0.1",
    "private": true,
    "scripts": {
        "start": "node node_modules/react-native/local-cli/cli.js start",
        "test": "jest",
        "react-devtools": "react-devtools",
        "precommit": "pretty-quick --staged",
        "flow start": "flow start",
        "flow stop": "flow stop",
        "flow status": "flow status",
        "flow coverage": "flow coverage"
    },
    "dependencies": {
        "autobind-decorator": "^2.1.0",
        "babel-plugin-transform-decorators-legacy": "^1.3.5",
        "prettysize": "^1.1.0",
        "prop-types": "^15.6.1",
        "react": "16.3.1",
        "react-native": "0.55.4",
        "react-native-action-button": "^2.8.4",
        "react-native-animatable": "^1.3.0",
        "react-native-circular-progress": "^0.2.0",
        "react-native-collapsible": "^0.12.0",
        "react-native-communications": "^2.2.1",
        "react-native-dev-menu": "^1.0.2",
        "react-native-drawer": "^2.5.0",
        "react-native-easy-toast": "^1.1.0",
        "react-native-elements": "^1.0.0-beta5",
        "react-native-fast-image": "^4.0.14",
        "react-native-i18n": "^2.0.12",
        "react-native-keychain": "^3.0.0-rc.3",
        "react-native-linear-gradient": "^2.4.0",
        "react-native-loading-spinner-overlay": "^0.5.2",
        "react-native-maps": "^0.21.0",
        "react-native-router-flux": "^3.43.0",
        "react-native-snap-carousel": "^3.7.2",
        "react-native-svg": "^6.3.1",
        "react-native-textinput-effects": "^0.4.2",
        "react-native-vector-icons": "^4.6.0",
        "react-redux": "^5.0.7",
        "redux": "^4.0.0",
        "redux-devtools-extension": "^2.13.2",
        "redux-thunk": "^2.3.0"
    },
    "devDependencies": {
        "babel-jest": "23.0.1",
        "babel-preset-react-native": "^5.0.2",
        "eslint-config-prettier": "^2.9.0",
        "eslint-config-rallycoding": "^3.2.0",
        "eslint-plugin-prettier": "^2.6.0",
        "husky": "^0.14.3",
        "jest": "23.1.0",
        "lint-staged": "^7.1.3",
        "prettier": "^1.13.4",
        "prettier-eslint": "^8.8.2",
        "pretty-quick": "^1.6.0",
        "react-devtools": "^3.2.3",
        "react-test-renderer": "16.3.1"
    },
    "jest": {
        "preset": "react-native"
    }
}

```

## Reproducible Demo
can not its private company code.

**Comment by yeomann** [comment]
I have added following in `package.json` after reading arround in issues. but still same error like above
```
 "@babel/core": "^7.0.0-beta.47",
   "babel-core": "^7.0.0-beta.47",
   "babel-preset-react-native": "^5.0.2",
```

**Comment by yeomann** [comment]
ok after some debugging around and fixing `node_modules` it seems that project is compiled successfully. well I have not done template changes yet for `0.56.0` so let me do that and check if everything is working or not.

**Comment by kelset** [comment]
We have clearly stated in the [Changelog](https://github.com/react-native-community/react-native-releases/blob/master/CHANGELOG.md#react-native-now-uses-babel-7) that you need to use fixed dependencies for both the preset and all the babel related dependencies.

Please refer to that paragraph and the links provided there to ensure that you are upgrading everything accordingly.

I'll close this issue until you can provide a repro that shows how this is an issue caused by React Native.

**Comment by yeomann** [comment]
Hi @kelset you can close for now but I can't provide you a repo sorry, the code is not open source. so, please don't ask me for a repo.
regarding your comment of 

> We have clearly stated in the Changelog that you need to use fixed dependencies for both the preset and all the babel related dependencies.

I did that clearly before trying to upgrade or running a upgrade command, As I explained above, Only ad only after make a upgrade of `babel`, I met the following error

```
git-upgrade info Configure Git environment
git-upgrade info Init Git repository
git-upgrade info Add all files to commit
git-upgrade info Commit current project sources
git-upgrade info Create a tag before updating sources
git-upgrade info Generate old version template
git-upgrade ERR! An error occurred during upgrade:
git-upgrade ERR! Error: Plugin 0 specified in "/Users/yeoman/Development/Xeront/ReactNative/ispApp/node_modules/babel-preset-react-native/index.js" provided an invalid property of "default" (While processing preset: "/Users/yeoman/Development/Xeront/ReactNative/ispApp/node_modules/babel-preset-react-native/index.js")
    at Plugin.init (/usr/local/lib/node_modules/react-native-git-upgrade/node_modules/babel-core/lib/transformation/plugin.js:131:13)
    at Function.normalisePlugin (/usr/local/lib/node_modules/react-native-git-upgrade/node_modules/babel-core/lib/transformation/file/options/option-manager.js:152:12)
    at /usr/local/lib/node_modules/react-native-git-upgrade/node_modules/babel-core/lib/transformation/file/options/option-manager.js:184:30
    at Array.map (<anonymous>)
    at Function.normalisePlugins (/usr/local/lib/node_modules/react-native-git-upgrade/node_modules/babel-core/lib/transformation/file/options/option-manager.js:158:20)
    at OptionManager.mergeOptions (/usr/local/lib/node_modules/react-native-git-upgrade/node_modules/babel-core/lib/transformation/file/options/option-manager.js:234:36)
    at /usr/local/lib/node_modules/react-native-git-upgrade/node_modules/babel-core/lib/transformation/file/options/option-manager.js:265:14
    at /usr/local/lib/node_modules/react-native-git-upgrade/node_modules/babel-core/lib/transformation/file/options/option-manager.js:323:22
    at Array.map (<anonymous>)
    at OptionManager.resolvePresets (/usr/local/lib/node_modules/react-native-git-upgrade/node_modules/babel-core/lib/transformation/file/options/option-manager.js:275:20)
git-upgrade ERR! Restore initial sources
Note: checking out 'project-snapshot'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 44b670a... Project snapshot
``` 

I am not sure either this will help or provide any assist regarding what cause the error. anyways after upgrading `react-native` and `react` manually, It seems there were some errors on `node_modules` I manually fix those errors and it seems build was a success!

**Comment by yeomann** [comment]
latest I have this error related with `react-native-maps`
```
:react-native-maps:prepareReleaseDependencies
Conflict with dependency 'com.android.support:support-v4' in project ':react-native-maps'. Resolved versions for compilation (26.1.0) and packaging (24.0.0) differ. This can generate runtime errors due to mismatched resources.
:react-native-maps:prepareReleaseDependencies FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':react-native-maps:prepareReleaseDependencies'.
> Dependency Error. See console for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

```

can anyone help me with some directions? or should I ask this in `react-native-maps` repo?

**Comment by kelset** [comment]
> I can't provide you a repo sorry, the code is not open source. so, please don't ask me for a repo.

A repRo is a "slang" term to indicate a small repository with the minimal amount of code to REPRODUCE the issue; it doesn't need to be your app source code, just a new `react-native init` project with the minimal changes necessary to create the issue described and a list of steps.

> I did that clearly before trying to upgrade or running a upgrade command, As I explained above,

In your previous comments you clearly used **not** fixed dependencies:
<img width="789" alt="screen shot 2018-07-18 at 14 46 54" src="https://user-images.githubusercontent.com/16104054/42885656-73128b4a-8a99-11e8-8e25-22fd570e2f62.png">

Fixed dependencies means, in this case:

```
 "@babel/core": "7.0.0-beta.47",
   "babel-core": "7.0.0-beta.47",
   "babel-preset-react-native": "5.0.2",
```

Meaning, removing the `^`.

> latest I have this error related with react-native-maps

As you can see in the error log:

> Conflict with dependency 'com.android.support:support-v4' in project ':react-native-maps'. Resolved versions for compilation (26.1.0) and packaging (24.0.0) differ. This can generate runtime errors due to mismatched resources.

You have a conflict between the gradle files options. This suggests that you have missconfigured your project android side config, have you checked this section of the [installation guide of maps](https://github.com/react-community/react-native-maps/blob/master/docs/installation.md#android)?

Anyway yes I think that if after fixing your gradle config you are still having react-native-maps issues you should open an issue there.

**Comment by yeomann** [comment]
thanks @kelset my project was working with react-native-maps before so map config was correct. installation guide from react native maps show grade 4 configs, and doesn't even say that those configs are of Gradle 4 😆  while I had few modification in `react-native-maps` under `node_modules/react-native-maps/lib/android/build.gradle` to make it run for grade 2 that react-native was using before 

I just only changed 

```
dependencies {
.....
.....
.....
  compileOnly "com.facebook.react:react-native:+"
  implementation "com.google.android.gms:play-services-base:$googlePlayServicesVersion"
  implementation "com.google.android.gms:play-services-maps:$googlePlayServicesVersion"
  implementation "com.google.maps.android:android-maps-utils:$androidMapsUtilsVersion"
}

TO BELOW

dependencies {
.....
......
......
  provided "com.facebook.react:react-native:+"
  compile "com.google.android.gms:play-services-base:$googlePlayServicesVersion"
  compile "com.google.android.gms:play-services-maps:$googlePlayServicesVersion"
  compile "com.google.maps.android:android-maps-utils:$androidMapsUtilsVersion"
}
```

anyways, I can read, it is a conflict issue, but I don't how to fix this because I don't have android/native knowledge yet. 

> Conflict with dependency 'com.android.support:support-v4' in project ':react-native-maps'. Resolved versions for compilation (26.1.0) and packaging (24.0.0) differ. This can generate runtime errors due to mismatched resources

anyways I don't know I guess I need to ask there. maybe if you know something enlighten me pls 😛

**Comment by yeomann** [comment]
anyways thanks again @kelset I don't I did something It fixed it. everything is working smoothly. cheers!
